pkg/clientcert/certificate.go

```diff
@@ -157,16 +166,81 @@ func BuildKubeconfig(clientConfig *restclient.Config, certPath, keyPath string)
 	return kubeconfig
 }
 
-// isCSRApproved returns true if the given csr has been approved
-func isCSRApproved(csr *certificatesv1.CertificateSigningRequest) bool {
+type csrControl interface {
+	get(name string) (metav1.Object, error)
+	create(ctx context.Context, recorder events.Recorder, objMeta metav1.ObjectMeta, csrData []byte, signerName string) (string, error)
+	isApproved(csr metav1.Object) bool
```

Comment:
should this be?

```
isApprovedandIssued(csrName string) ([]byte, bool, error)

```

and we do not need to expose `get`

Reply:
aesthetically i think separating isApproved and isIssued will look cleaner, but i eliminated getter from interface wdyt?